.github/workflows/deploy_prod.yaml

```diff
@@ -6,34 +6,103 @@ on:
-          --project=curator-348003 \
-          --region=australia-southeast1 \
-          --service-account=curator-prod@curator-348003.iam.gserviceaccount.com \
-          --no-allow-unauthenticated \
-          --add-cloudsql-instances=curator-348003:australia-southeast1:curator-postgres \
-          --source=. \
-          --port=8000 \
-          --set-env-vars=DJANGO_SETTINGS_MODULE=curation_portal.settings.base,ALLOWED_HOSTS=curator.populationgenomics.org.au,CURATION_PORTAL_AUTH_HEADER=HTTP_X_GOOG_AUTHENTICATED_USER_EMAIL,DB_ENGINE=django.db.backends.postgresql,DB_HOST=/cloudsql/curator-348003:australia-southeast1:curator-postgres,DB_DATABASE=curator-prod,DB_USER=curator-prod \
-          --update-secrets=SECRET_KEY=django-secret-key-prod:latest,DB_PASSWORD=postgres-password-prod:latest \
-          curator-prod
+      - name: "checkout repo"
+        uses: actions/checkout@v2
+
+      - name: "gcloud setup"
+        uses: google-github-actions/setup-gcloud@v0
+        with:
+          project_id: curator-348003
+          service_account_key: ${{ secrets.GCP_DEPLOY_KEY }}
+
+      - name: 'gcloud docker auth'
+        run: |
+          gcloud auth configure-docker australia-southeast1-docker.pkg.dev
+
+      - name: "set short SHA"
+        id: repo
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+   
+      - name: 'build image'
+        run: |
+          docker build -t australia-southeast1-docker.pkg.dev/curator-348003/images/curator-prod:${{ steps.repo.outputs.short_sha }} . --no-cache
+      
+      - name: 'push image'
+        run: |
+          docker push $DOCKER_IMAGE
+
+  migrate:
+    runs-on: ubuntu-latest
+
+    env:
+      CLOUDSDK_CORE_DISABLE_PROMPTS: 1
+
+    steps:
+      - name: "checkout repo"
+        uses: actions/checkout@v2
+
+      - name: "gcloud setup"
+        uses: google-github-actions/setup-gcloud@v0
+        with:
+          project_id: curator-348003
+          service_account_key: ${{ secrets.GCP_DEPLOY_KEY }}
+
+      - name: "set short SHA"
+        id: repo
+        run: echo "sha_short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: "build and deploy"
+        run: |
+          gcloud beta run jobs create \
+            curator-prod-migrate-db \
+            --region=australia-southeast1 \
+            --image=australia-southeast1-docker.pkg.dev/curator-348003/images/curator-prod:${{ steps.repo.outputs.short_sha }} \
+            --service-account=curator-prod@curator-348003.iam.gserviceaccount.com \
+            --set-cloudsql-instances=curator-348003:australia-southeast1:curator-postgres \
+            --set-env-vars=DJANGO_SETTINGS_MODULE=curation_portal.settings.base,DB_ENGINE=django.db.backends.postgresql,DB_HOST=/cloudsql/curator-348003:australia-southeast1:curator-postgres,DB_DATABASE=curator-prod,DB_USER=curator-prod \
+            --set-secrets=SECRET_KEY=django-secret-key-prod:latest,DB_PASSWORD=postgres-password-prod:latest \
```

Comment:
Can we factor out the common arguments for this and the `deploy` job below?

Reply:
Updated and merge into a single job to share info. I've also added a step to set deployment information (database settings, settings module, service account etc) based on the branch to support prod and dev deployments.